kafka-connect-http/src/main/java/com/github/castorm/kafka/connect/http/HttpSourceTask.java

```diff
@@ -104,11 +117,37 @@ private Offset loadOffset(Map<String, String> initialOffset) {
 
         throttler.throttle(offset.getTimestamp().orElseGet(Instant::now));
 
-        HttpRequest request = requestFactory.createRequest(offset);
+//        HttpRequest request = requestFactory.createRequest(offset);
+
+        List<SourceRecord> records = new ArrayList<>();
+
+        if(handlePagination && !Objects.isNull(modifiedUrl)) {
+            request = HttpRequest.builder()
+                .method(request.getMethod())
+                .url(modifiedUrl)
+                .headers(request.getHeaders())
+                .body(request.getBody())
+                .build();
```

Comment:
We are creating a request here. Why not doing it in the entity meant for that: `RequestFactory`?

You might want to create a `RequestFactory` specific for this use case though.

Reply:
I tried to create a different request factory to take care of pagination. But I am stuck at a point. I am not able to understand how to pass the next page URL from the response to this request factory to create a new request.